data/mods/megasforall/abilities.ts

```diff
@@ -1640,55 +1638,33 @@ export const Abilities: {[abilityid: string]: ModdedAbilityData} = {
 					pokemon.side.pokemon[i].getAbility().isPermanent || additionalBannedAbilities.includes(pokemon.side.pokemon[i].ability)
 				) {
 					continue;
-				} else {
-					break;
 				}
+				break;
 			}
-			if (!pokemon.side.pokemon[i]) return;
-			if (pokemon === pokemon.side.pokemon[i]) return;
-			const masquerade = pokemon.side.pokemon[i];
-			this.add('-ability', pokemon, 'Masquerade');
-			pokemon.setAbility(masquerade.ability);
-			this.add('-message', `${pokemon.name} inherited ${this.dex.getAbility(pokemon.ability).name} from ${masquerade.name}!`);
-			this.add('-ability', pokemon, this.dex.getAbility(pokemon.ability).name, '[silent]');
-		},
-		onSwitchIn(pokemon) {
-			pokemon.addVolatile('masquerade');
-			let i;
-			for (i = pokemon.side.pokemon.length - 1; i > pokemon.position; i--) {
-				if (!pokemon.side.pokemon[i]) continue;
-				const additionalBannedAbilities = [
-					'noability', 'flowergift', 'forecast', 'hugepower', 'hungerswitch', 'illusion', 'imposter', 'neutralizinggas',
-					'powerofalchemy', 'purepower', 'receiver', 'trace', 'wonderguard',
-				];
-				if (
-					pokemon.side.pokemon[i].fainted ||
-					pokemon.side.pokemon[i].getAbility().isPermanent || additionalBannedAbilities.includes(pokemon.side.pokemon[i].ability)
-				) {
-					continue;
-				} else {
-					break;
-				}
+			if (!pokemon.side.pokemon[i] || pokemon === pokemon.side.pokemon[i]) {
+				this.effectData.gaveUp = true;
+				return;
 			}
-			if (!pokemon.side.pokemon[i]) return;
-			if (pokemon === pokemon.side.pokemon[i]) return;
 			const masquerade = pokemon.side.pokemon[i];
 			this.add('-ability', pokemon, 'Masquerade');
 			pokemon.setAbility(masquerade.ability);
-			this.add('-message', `${pokemon.name} inherited ${this.dex.getAbility(pokemon.ability).name} from ${masquerade.name}!`);
+			this.add('-message', `(${pokemon.name} inherited ${this.dex.getAbility(pokemon.ability).name} from ${masquerade.name}!)`);
```

Comment:
```suggestion
			this.hint(${pokemon.name} inherited ${this.dex.getAbility(pokemon.ability).name} from ${masquerade.name}!`);
```

Reply:
(I think this might still need the ` at the front, so I'm going to commit this in and then also fix that! Let me know if that's incorrect, though!
But yeah, doing this.hint makes total sense - thanks!!)